### PR TITLE
Add Related Filters to UserFilterSet

### PR DESCRIPTION
**Pull request by egenius01**
# Closes: #<ISSUE NUMBER GOES HERE>
# What's Changed

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/users/tests/test_filters.py [hunk 1]
```diff
@@ -13,19 +13,90 @@
...
+def common_test_data(cls):
+    pass
+
+    # users = (
+    #     User.objects.create_user(username="TestCaseUser 1"),
+    #     User.objects.create_user(username="TestCaseUser 2"),
+    #     User.objects.create_user(username="TestCaseUser 3"),
+    # )
+
+
```
Comment (bryanculver) [inline nautobot/users/tests/test_filters.py:93]: It seems this function serves no purpose.

File: nautobot/users/filters.py [hunk 2]
```diff
@@ -52,6 +88,19 @@ class Meta:
             "email",
             "is_staff",
             "is_active",
+            "config_data",
+            "changes",
+            "has_changes",
```
Comment (bryanculver) [inline nautobot/users/filters.py:93]: If you have the field defined above it is not needed here. This section is for "magic" filters that introspect fields on the given model and provide determined proper filter and lookup expression combinations.

**Review by bryanculver — CHANGES_REQUESTED** [review]
There is some cleanup that should be done with regard to commented-out code and functions that do nothing.

Also, please update the PR description with the planned/desired use cases for these changes.

File: nautobot/users/filters.py [hunk 3]
```diff
@@ -52,6 +88,19 @@ class Meta:
...
+            # "logentry_set",
+            # "has_logentry_set",
```
[the remark this answers is not in the document]
Comment (egenius01) [reply nautobot/users/filters.py:95]: Alright i'd do that, Thanks

**Pull request closed without merging**